### PR TITLE
[D3DX9_24 => 43] Sync with Wine Staging 2.16. CORE-13762

### PR DESCRIPTION
**Pull request by AmineKhaldi**

### CODE REVIEW

File: dll/directx/wine/d3dx9_36/d3dx9_36_private.h [hunk 1]
```diff
@@ -125,8 +143,268 @@ const char *debug_d3dxparameter_type(D3DXPARAMETER_TYPE t) DECLSPEC_HIDDEN;
...
+static inline BOOL is_param_dirty(struct d3dx_parameter *param, ULONG64 update_version)
+{
+    return is_top_level_param_dirty(param->top_level_param, update_version);
+}
+
+struct d3dx_parameter *get_parameter_by_name(struct d3dx9_base_effect *base,
+        struct d3dx_parameter *parameter, const char *name) DECLSPEC_HIDDEN;
+
+#ifdef __REACTOS__
+#define SET_D3D_STATE_(_manager, _device, _method, ...) ((_manager) ? (_manager)->lpVtbl->_method((_manager), __VA_ARGS__) \
```
Comment (ThFabba) [inline dll/directx/wine/d3dx9_36/d3dx9_36_private.h]: Interesting. I'm curious what made the underscores and parentheses necessary.

File: dll/directx/wine/d3dx9_36/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
-add_definitions(-D__WINESRC__ -Dcopysignf=_copysignf)
+add_definitions(-D__WINESRC__ -Dcopysignf=_copysignf -D_D3DX9_VER=36)
```
Comment (ThFabba) [inline dll/directx/wine/d3dx9_36/CMakeLists.txt]: This is a little bit scary. Does that mean Wine can have different code between the different version, but we'll just keep forwarding?

File: dll/directx/wine/d3dx9_36/d3dx9_36_private.h (hunk 1 above)
Comment (AmineKhaldi) [reply dll/directx/wine/d3dx9_36/d3dx9_36_private.h]: We hit a compile error with MSVC:
```error C2063: 'element_count' : not a function```

on lines such as:
```c
return SET_D3D_STATE(effect, SetVertexShaderConstantF, index, (const float *)value_ptr, element_count);
```

File: dll/directx/wine/d3dx9_36/CMakeLists.txt (hunk 2 above)
Comment (AmineKhaldi) [reply dll/directx/wine/d3dx9_36/CMakeLists.txt]: Yeap, I'm afraid eventually we'll have to compile the code over and over for each module like upstream when the bugs appear and become a priority.

Comment (tkreuzer) [reply dll/directx/wine/d3dx9_36/CMakeLists.txt]: It is currently only used to extend a vTable. Using maximum is fine.

**Pull request merged**